server/src/test/java/org/elasticsearch/reservedstate/service/FileSettingsServiceTests.java

```diff
@@ -352,15 +355,22 @@ public void testHandleSnapshotRestoreResetsMetadata() throws Exception {
     }
 
     // helpers
-    private void writeTestFile(Path path, String contents) throws IOException {
-        Path tempFilePath = createTempFile();
-        Files.writeString(tempFilePath, contents);
-        Files.move(tempFilePath, path, StandardCopyOption.ATOMIC_MOVE, StandardCopyOption.REPLACE_EXISTING);
-    }
+    private static void writeTestFile(Path path, String contents) {
```

Comment:
Tweaking this to:

- account for missing atomic_move functionality
- make temp file removal more explicit